### DISCUSSION TITLE
How can I send a signal with data of a custom datatype?

### DISCUSSION THREAD

**Discussion by derived-coder** [Kistler-Group/sdbus-cpp#240] [Q&A]
Hi, do you have an example how I can send a custom datatype as argument to a signal?
What do I need to define in my config file?

```
 <signal name="propertyChanged">
      <arg type="CustomType?" name="property" />
    </signal>
```

**Comment by Oxymoron79** [comment]
The `type` string must comply to the [D-Bus type system specification](https://dbus.freedesktop.org/doc/dbus-specification.html#type-system). But the type system is not restricted to basic types (int, string, etc.), it also allows to compose complex types with containers (array, struct, dict, etc.). That should suffice to express any `CustomType`.

I hope this helps.
